eth/core/processor_test.go

```diff
@@ -44,73 +44,52 @@ var (
 	signer        = types.LatestSignerForChainID(params.DefaultChainConfig.ChainID)
 	_             = key
 	_             = signer
+	blockGasLimit = 10000000
 	dummyHeader   = &types.Header{
 		Number:   big.NewInt(1),
-		GasLimit: 1000000,
+		BaseFee:  big.NewInt(1),
+		GasLimit: uint64(blockGasLimit),
 	}
 	legacyTxData = &types.LegacyTx{
 		Nonce:    0,
 		To:       &dummyContract,
-		Gas:      100000,
-		GasPrice: big.NewInt(2),
+		Gas:      1000000,
+		GasPrice: big.NewInt(1),
 		Data:     []byte("abcdef"),
 	}
 )
 
 var _ = Describe("StateProcessor", func() {
 	var (
-		sdb           *vmmock.PolarisStateDBMock
-		host          *mock.PolarisHostChainMock
-		bp            *mock.BlockPluginMock
-		gp            *mock.GasPluginMock
-		cp            *mock.ConfigurationPluginMock
-		pp            *mock.PrecompilePluginMock
-		sp            *core.StateProcessor
-		blockGasLimit uint64
+		sdb *vmmock.PolarisStateDBMock
+		bp  *mock.BlockPluginMock
+		gp  *mock.GasPluginMock
+		cp  *mock.ConfigurationPluginMock
+		pp  *mock.PrecompilePluginMock
+		sp  *core.StateProcessor
+		evm *vm.GethEVM
 	)
 
 	BeforeEach(func() {
 		sdb = vmmock.NewEmptyStateDB()
-		host = mock.NewMockHost()
-		bp = mock.NewBlockPluginMock()
-		gp = mock.NewGasPluginMock()
-		cp = mock.NewConfigurationPluginMock()
-		pp = mock.NewPrecompilePluginMock()
-		host.GetBlockPluginFunc = func() core.BlockPlugin {
-			return bp
-		}
-		host.GetGasPluginFunc = func() core.GasPlugin {
-			return gp
-		}
-		host.GetConfigurationPluginFunc = func() core.ConfigurationPlugin {
-			return cp
-		}
-		host.GetPrecompilePluginFunc = func() core.PrecompilePlugin {
-			return pp
-		}
-		pp.RegisterFunc = func(pc vm.PrecompileContainer) error {
-			return nil
-		}
-		sp = core.NewStateProcessor(cp, gp, pp, sdb, &vm.Config{})
-		Expect(sp).ToNot(BeNil())
-		blockGasLimit = 1000000
-
+		_, bp, cp, gp, _, pp, _, _ = mock.NewMockHostAndPlugins()
 		bp.GetNewBlockMetadataFunc = func(n int64) (common.Address, uint64) {
 			return common.BytesToAddress([]byte{2}), uint64(3)
 		}
 		pp.HasFunc = func(addr common.Address) bool {
 			return false
 		}
+		gp.SetBlockGasLimit(uint64(blockGasLimit))
 
-		sdb.PrepareFunc = func(rules params.Rules, sender common.Address,
-			coinbase common.Address, dest *common.Address,
-			precompiles []common.Address, txAccesses types.AccessList,
-		) {
-			// no-op
-		}
-
-		gp.SetBlockGasLimit(blockGasLimit)
-		sp.Prepare(context.Background(), nil, dummyHeader)
+		sp = core.NewStateProcessor(cp, gp, pp, sdb, &vm.Config{})
+		Expect(sp).ToNot(BeNil())
+		evm = vm.NewGethEVMWithPrecompiles(
+			vm.BlockContext{
```

Comment:
don't need to do this, there is a function that'll build you a context, saves the Transfer and CanTransfer imports.